controllers/userController.js

```diff
@@ -147,4 +148,41 @@ const deleteUserAddresses = async (req, res) => {
     res.status(500).send({ message: err })
   }
 }
-module.exports = { userRegister, login, getUserData, deleteUserData, limitUsersData, userAddress, deleteUserAddresses }
+
+const passwordResetToken = async (req, res) => {
+  const { user_id } = req.body
+  try {
+    const userData = await userModel.findOne({ where: { id: user_id } })
+    if (userData) {
+      const passwordResetToken = jwt.sign({
+        data: `${userData.user_id} ${userData.password}`
+      }, process.env.jwtSecKey, { expiresIn: 60 * 10 });
+      await userModel.update({ jwtToken: passwordResetToken }, { where: { id: user_id } })
+      res.status(200).send(passwordResetToken)
+
+    } else {
+      res.status(404).send({ message: "user not found, please check the user id" })
+    }
+
+  } catch (err) { res.status(500).send(err) }
+
+}
+const verifyResetPassword = async (req, res) => {
+  const resetToken = req.params.passwordResetToken
+  const { user_id, password } = req.body
+  try {
+    const userData = await userModel.findOne({ where: { id: user_id } })
+    const verifyToken = jwt.verify(resetToken, process.env.jwtSecKey)
+    if (userData.jwtToken === resetToken && verifyToken) {
```

Comment:
and also here you can not directly reset the password if the token is valid than take a password if password is correct then create a hashpassword and update in the password field.